integration_tests/src/main/python/prune_partition_column_test.py

```diff
@@ -127,3 +128,78 @@ def test_prune_partition_column_when_filter_fallback_project(spark_tmp_path, pru
                                                              filter_col, file_format):
     do_prune_partition_column_when_filter_project(spark_tmp_path, prune_part_enabled, file_format,
                                                   filter_col, gpu_project_enabled=False)
+
+def create_contacts_table_and_read(is_partitioned, format, data_path, expected_schemata, func, conf):
+    full_name_type = StructGen([('first', StringGen()), ('middle', StringGen()), ('last', StringGen())])
+    name_type = StructGen([('first', StringGen()), ('last', StringGen())])
+    contacts_data_gen = StructGen([
```

Comment:
Is there a reason you didn't include all of the columns from the SchemaPruningSuite test?

You are missing relatives, employer, and relations from contacts_data_gen.

brief contacts looks okay, but there were other tables in the file too that are not covered here.

Reply:
I didn't include them because they weren't being used in the projection or criteria. 
Was that not a fair conclusion? Do you think we should include them either way?

Reply:
> Was that not a fair conclusion? Do you think we should include them either way?

I think it is alright to drop some of the columns from the `Contacts` table in `SchemaPruningSuite.scala`. The reason to keep them would be to have columns in the schema that are pruned out. It appears that we've retained enough columns for that.

For instance, `SELECT friends.middle, friends` verifies that `id`, `name`, and `address` have been pruned out.

Reply:
I agree. My main concern is that there are other tests in the same file that we are not covering with this. I wasn't sure if we wanted to include everything assuming that they would be added later on or not. Just wanted to understand the logic behind it.  Might also be good if we are referencing a specific spark test, like we do here, that we include a comment explaining it. Just in case someone else comes along and does try to pull over more tests or update this to match changes to the tests in Spark.